shell/platform/windows/compositor_opengl.cc

```diff
@@ -0,0 +1,150 @@
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/compositor_opengl.h"
+
+#include "GLES3/gl3.h"
+#include "flutter/shell/platform/windows/flutter_windows_view.h"
+
+namespace flutter {
+
+namespace {
+
+// The metadata for an OpenGL framebuffer backing store.
+struct FramebufferBackingStore {
+  uint32_t framebuffer_id;
+  uint32_t texture_id;
+};
+
+}  // namespace
+
+CompositorOpenGL::CompositorOpenGL(FlutterWindowsEngine* engine,
+                                   impeller::ProcTableGLES::Resolver resolver)
+    : engine_(engine), resolver_(resolver) {}
+
+bool CompositorOpenGL::CreateBackingStore(
+    const FlutterBackingStoreConfig& config,
+    FlutterBackingStore* result) {
+  if (!is_initialized_ && !Initialize()) {
+    return false;
+  }
+
+  auto store = std::make_unique<FramebufferBackingStore>();
+
+  gl_->GenTextures(1, &store->texture_id);
+  gl_->GenFramebuffers(1, &store->framebuffer_id);
+
+  gl_->BindFramebuffer(GL_FRAMEBUFFER, store->framebuffer_id);
+
+  gl_->BindTexture(GL_TEXTURE_2D, store->texture_id);
+  gl_->TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+  gl_->TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+  gl_->TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+  gl_->TexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+  gl_->TexImage2D(GL_TEXTURE_2D, 0, GL_RGBA8, config.size.width,
+                  config.size.height, 0, GL_RGBA, GL_UNSIGNED_BYTE, nullptr);
+  gl_->BindTexture(GL_TEXTURE_2D, 0);
+
+  gl_->FramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0_EXT,
+                            GL_TEXTURE_2D, store->texture_id, 0);
+
+  result->type = kFlutterBackingStoreTypeOpenGL;
+  result->open_gl.type = kFlutterOpenGLTargetTypeFramebuffer;
+  result->open_gl.framebuffer.name = store->framebuffer_id;
+  result->open_gl.framebuffer.target = format_;
+  result->open_gl.framebuffer.user_data = store.release();
+  result->open_gl.framebuffer.destruction_callback = [](void* user_data) {
+    // Backing store destroyed in `CompositorOpenGL::CollectBackingStore`, set
+    // on FlutterCompositor.collect_backing_store_callback during engine start.
+  };
+  return true;
+}
+
+bool CompositorOpenGL::CollectBackingStore(const FlutterBackingStore* store) {
+  FML_DCHECK(is_initialized_);
+  FML_DCHECK(store->type == kFlutterBackingStoreTypeOpenGL);
+  FML_DCHECK(store->open_gl.type == kFlutterOpenGLTargetTypeFramebuffer);
+
+  auto user_data = static_cast<FramebufferBackingStore*>(
+      store->open_gl.framebuffer.user_data);
+
+  gl_->DeleteFramebuffers(1, &user_data->framebuffer_id);
+  gl_->DeleteTextures(1, &user_data->texture_id);
+
+  delete user_data;
+  return true;
+}
+
+bool CompositorOpenGL::Present(const FlutterLayer** layers,
+                               size_t layers_count) {
+  FML_DCHECK(is_initialized_);
+  FML_DCHECK(layers_count == 1);
```

Comment:
Maybe add a TODO here to explain that composition is not yet supported? As well as the assertion for the layer below.

Reply:
Added TODO!